qt/aqt/editor.py

```diff
@@ -242,38 +242,35 @@ def addButton(
         rightside: bool = True,
     ) -> str:
         """Assign func to bridge cmd, register shortcut, return button"""
-        if func:
 
-            def wrapped_func(editor: Editor) -> None:
-                self.call_after_note_saved(
-                    functools.partial(func, editor), keepFocus=True
-                )
+        def wrapped_func(editor: Editor) -> None:
+            self.call_after_note_saved(functools.partial(func, editor), keepFocus=True)
 
-            self._links[cmd] = wrapped_func
+        self._links[cmd] = wrapped_func
 
-            if keys:
+        if keys:
 
-                def on_activated() -> None:
-                    wrapped_func(self)
+            def on_activated() -> None:
+                wrapped_func(self)
 
-                if toggleable:
-                    # generate a random id for triggering toggle
-                    id = id or str(randrange(1_000_000))
+            if toggleable:
+                # generate a random id for triggering toggle
+                id = id or str(randrange(1_000_000))
 
-                    def on_hotkey() -> None:
-                        on_activated()
-                        self.web.eval(
-                            f'toggleEditorButton(document.getElementById("{id}"));'
-                        )
+                def on_hotkey() -> None:
+                    on_activated()
+                    self.web.eval(
+                        f'toggleEditorButton(document.getElementById("{id}"));'
+                    )
 
-                else:
-                    on_hotkey = on_activated
+            else:
+                on_hotkey = on_activated
 
-                QShortcut(  # type: ignore
-                    QKeySequence(keys),
-                    self.widget,
-                    activated=on_hotkey,
-                )
+            QShortcut(  # type: ignore
+                QKeySequence(keys),
+                self.widget,
+                activated=on_hotkey,
+            )
```

Comment:
I moved this code out of the if-block because the mypy error was `qt/aqt/editor.py:245: error: Function "func" could always be true in boolean context  [truthy-function]`

I think an alternative could be changing the type in the function definition to `func: Callable[[Editor], None] | None`

Reply:
I'd make the same call :+1: